tests/test_other.py

```diff
@@ -446,3 +446,63 @@ def test_targeting_trainable_tokens_raises(self):
         msg = "trainable_token_indices cannot be applied to modules of type <class 'peft.tuners.lora.layer.Embedding'>"
         with pytest.raises(TypeError, match=msg) as e:
             get_peft_model(model, peft_config)
+
+
+class TestAdapterTargeting:
+    """Make sure that already existing adapters cannot be targeted to avoid conflicts."""
+
+    @pytest.fixture
+    def base_model_cls(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.l1 = torch.nn.Linear(10, 20)
+                self.l2 = torch.nn.Conv2d(1, 1, 2)
+
+            def forward(self, x):
+                return self.l2(self.l1(x))
+
+        return M
+
+    @pytest.mark.parametrize(
+        "config_cls, config_kwargs",
+        [
+            (LoraConfig, {"target_modules": "l1.*"}),
+            (LoraConfig, {"target_modules": "l2.*"}),
+            (VeraConfig, {"target_modules": "l1.*"}),
+            (VeraConfig, {"target_modules": "(l1|vera_A).*"}),  # also target the shared layer
```

Comment:
Guessing we don't have to be exhaustive in the config classes to check this with?

Reply:
Yeah, things like targeting `trainable_tokens` and `modules_to_save` are tested separately which leaves `target_modules`. As far as I can tell most (all?) adapter implementations base their `target_modules` off of `BaseTuner.inject_adapter`, so I thought it sufficient to test with LoRA for the base case and VeRA for when shared parameters are introduced.